tests/DeprecatedTestMethods.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests;
+
+/**
+ * @internal
+ *
+ * @method static assertMatchesRegularExpression($pattern, $string, $message = '')
+ * @method static assertFileDoesNotExist(...$args)
+ * @method static assertAttributeSame($expected, $attribute, $object)
+ * @method        expectExceptionMessageMatches($message)
+ * @method        expectExceptionMessageRegExp($regexp)
+ * @method static getStaticAttribute($object, $attribute)
+ */
+trait DeprecatedTestMethods
+{
+    /**
+     * This method will only trigger under PHP <8.
+     *
+     * @param string $name
+     * @param mixed  $arguments
+     */
+    public static function __callStatic($name, $arguments)
+    {
+        if ('assertMatchesRegularExpression' === $name) {
+            self::assertRegExp(...$arguments);
+
+            return;
+        }
+
+        if ('assertFileDoesNotExist' === $name) {
+            self::assertFileNotExist(...$arguments);
+        }
+
+        if ('assertAttributeSame' === $name) {
+            self::assertAttributeSameActual(...$arguments);
+
+            return;
+        }
+
+        if ('assertAttributeSame' === $name) {
+            self::assertAttributeSameActual(...$arguments);
+
+            return;
+        }
+
+        if ('getObjectAttribute' === $name) {
+            return self::getObjectAttributeActual(...$arguments);
+        }
+
+        if ('getStaticAttribute' === $name) {
+            return self::getStaticAttributeActual(...$arguments);
+        }
+    }
+
+    public function __call($name, $arguments)
+    {
+        if ('expectExceptionMessageMatches' === $name) {
+            $this->expectExceptionMessageRegExp(...$arguments);
+
+            return;
+        }
```

Comment:
this will silence all the errors like `$this->asdfasfasfdadfsfasdf()`.

I would suggest to create real `expectExceptionMessageRegExp` method or at least throw the exception as last step of `__call`

Reply:
Aye, I'll be adding these methods `PHPUnitGoodPractices/polyfill`.